security/kernel-lockdown-tests.py

```diff
@@ -0,0 +1,96 @@
+#!/usr/bin/env python
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright: 2023 IBM.
+# Author: Nageswara R Sastry <rnsastry@linux.ibm.com>
+
+import os
+from avocado import Test
+from avocado.utils import distro, genio, linux_modules, process
+
+
+class kernelLockdown(Test):
+    """
+    Kernel Lockdown tests for Linux
+    :avocado: tags=privileged,security,lockdown
+    """
+
+    def setUp(self):
+        '''
+        Kernel lockdown and Guest Secure boot support is available only on P10.
+        '''
+        self.distro_version = distro.detect()
+        val = genio.read_file("/proc/cpuinfo")
+        if 'POWER10' not in val:
+            self.cancel("Power10 LPAR is required to run this test.")
+        # Checking whether lockdown enabled or not.
+        self.lockdown_file = "/sys/kernel/security/lockdown"
+        lockdown = genio.read_file(self.lockdown_file).rstrip('\t\r\n\0')
+        index1 = int(lockdown.index('[')) + 1
+        index2 = int(lockdown.index(']'))
+        sys_lockdown = lockdown[index1:index2]
+        if sys_lockdown == 'none':
+            self.cancel("Lockdown not enabled, can't execute test(s)")
+        # Checking the Guest Secure Boot enabled or not.
+        cmd = "lsprop  /proc/device-tree/ibm,secure-boot"
```

Comment:
is lsporp installed by default, is not worth to check if it installed ?

Reply:
resolved by adding package install check.